### PR TITLE
LOG-4805: The log severity should be sorted from most priority to less

### PR DESCRIPTION
**Pull request by Clee2691**
### Description
This PR changes the way a log event's `level` is set. It will first match on well known, structured patterns and if none match, it will match on unstructured keywords.

The order of the level is as follows and is sorted in descending priority: `emergency`, `alert`, `critical`, `error`, `warn`, `notice`, `info`, `debug`.

/cc @cahartma @vparfonov 
/assign @jcantrill 

### Links

- JIRA: https://issues.redhat.com/browse/LOG-4805

### CODE REVIEW

**Review by jcantrill — APPROVED** [review]
/approve

File: test/functional/normalization/loglevel/container_logs_test.go [hunk 1]
```diff
@@ -57,7 +58,6 @@ var _ = Describe("[functional][normalization][loglevel] tests for message format
 		outputTestLog := logs[0]
 		Expect(outputTestLog).To(FitLogFormatTemplate(outputLogTemplate))
 	},
-		Entry("should recognize a WARN message", "warn", "Warning: failed to query journal: Bad message OS Error 74"),
```
Comment (vparfonov) [inline test/functional/normalization/loglevel/container_logs_test.go:60]: Why do you remove this test?

File: test/functional/normalization/loglevel/container_logs_test.go
Comment (vparfonov) [inline test/functional/normalization/loglevel/container_logs_test.go:1]: This is not related to your changes, but can you take a look at the tests from line 65 and below, look like description not fit to given parameters

File: test/functional/normalization/loglevel/container_logs_test.go (hunk 1 above)
Comment (Clee2691) [reply test/functional/normalization/loglevel/container_logs_test.go:60]: We determined that we would stick with the changes for the keyword order and just remove this specific test. This is our best effort for setting log severity.

**Pull request merged**